tests/operators/models/test_transient_inversion_recovery.py

```diff
@@ -0,0 +1,93 @@
+"""Tests for the Transient Inversion Recovery model."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+from mrpro.operators.models import TransientInversionRecovery
+from tests import RandomGenerator
+
+
+def create_data(other=10, coils=5, z=100, y=100, x=100):
+    random_generator = RandomGenerator(seed=0)
+    m0 = random_generator.complex64_tensor(size=(other, coils, z, y, x))
+    t1 = random_generator.float32_tensor(size=(other, coils, z, y, x), low=1e-10)
+    alpha = random_generator.float32_tensor(size=(other, coils, z, y, x), low=3, high=9)
+    return m0, t1, alpha
+
+
+@pytest.mark.parametrize(
+    ('t', 'result'),
+    [
+        # short ti
+        (0, '-m0'),
+        # long ti
+        (20, 'm0_star'),
+    ],
+)
+def test_signal_boundaries_single_inversion(t, result):
```

Comment:
can we add a check for the signal at t1=0, t1=inf, m0=0, m0=inf.
it should not return nans.

Also, it would already be nice to add tests for the gradients.
Or, instead, I can add the tests for each signal model during the hackathon.
If the later, I would only add tests. If one of the tests fails, I will then open PRs for each failing signal function for somebody else to fix it...

Reply:
> If the later, I would only add tests. If one of the tests fails, I will then open PRs for each failing signal function for somebody else to fix it...

sounds like a good plan!